File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.Arrays;
import java.util.stream.Collectors;

class ResistorColor {
    enum Colours {
        black,
        brown,
        red,
        orange,
        yellow,
        green,
        blue,
        violet,
        grey,
        white
    }
    int colorCode(String color) {
        return Colours.valueOf(color).ordinal();
    }

    String[] colors() {
        return Arrays.stream(Colours.values()).map(Enum::name).toArray(String[]::new);
    }
}

File: zig/resistor-color/resistor_color.zig
const std = @import("std");
pub const ColorBand = enum { black, brown, red, orange, yellow, green, blue, violet, grey, white };
pub fn colorCode(color: ColorBand) isize {
    return @enumToInt(color);
}
pub fn colors() []const ColorBand {
    return std.enums.values(ColorBand);
}
